### PR TITLE
Support environment variable configuration for commands

### PR DESCRIPTION
**Pull request by ajkolean**
Resolves https://github.com/tuist/tuist/issues/6031

### Short description 📝

This PR introduces the capability to configure Tuist commands via environment variables, enhancing automation and flexibility, particularly in CI/CD environments. By exposing command parameters as environment variables prefixed with `TUIST_`, users can streamline their workflow and ensure consistency across different environments.

### Key features:

- **Environment Variable Support**: Allows setting all CLI options, flags, and arguments via environment variables prefixed with `TUIST_`.
- **Precedence**: CLI arguments take precedence over environment variables, which in turn take precedence over the defaults (if set).
- **Boolean Flag Handling**: All boolean flags have been given an inverted case, allowing environment variable overrides without conflicting with CLI argument options.
- **Enhanced CI/CD Automation**: Simplifies configuration of Tuist commands in CI/CD pipelines, ensuring consistent behavior without manual intervention.
- **Improved Help Output**: Includes environment variable names in the help output for better discoverability.
- **Documentation Generation**: Adds a script that generates CLI command reference documentation from each command's man page and converts it to Markdown.
- **Environment Variable Naming Convention**: The environment variables follow the format `[super.command.name...][command.name][argumentname]`

### How to test the changes locally 🧐

1. Set the desired environment variables in your shell. For example:
   ```sh
   export TUIST_BUILD_OPTIONS_SCHEME=MyScheme
   export TUIST_TEST_CLEAN=true
   export TUIST_PLUGIN_BUILD_PRODUCTS=Product1,Product2
   ```
2. Run the relevant Tuist command, such as `tuist generate`.
3. Verify that the command behaves as expected, with the parameters set by the environment variables.
4. Test various combinations of the introduced environment variables to ensure they are correctly parsed and applied.
5. Check the help output:
   ```sh
   swift run tuist help build
   TUIST_BUILD_BINARY_CACHE=false swift run tuist help build
   ```
   *Note that the help output now includes the environment key, making it easy to find, and reflects the default value that is used. The default will be the system default or, if an environment key is specified, it will reflect that.*

   Notice how setting the environment variable in the above example changes the default shown in the help from `binary-cache` to `--no-binary-cache`.

6. To see the generated CLI documentation, run the following commands:
   ```sh
   mise run docs:generate-cli-docs
   mise run docs:generate-manifests-docs
   mise run docs:dev
   ```

### Contributor checklist ✅

- [x] The code has been linted using `run mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Commands/EnvKey/EnvKey.swift [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    case cloudProjectListPath = "TUIST_CLOUD_PROJECT_LIST_PATH"
+
+    // CLOUD ORGANIZATION UPDATE MEMBER
+    case cloudOrganizationUpdateMemberOrganizationName = "TUIST_CLOUD_ORGANIZATION_UPDATE_MEMBER_ORGANIZATION_NAME"
+    case cloudOrganizationUpdateMemberUsername = "TUIST_CLOUD_ORGANIZATION_UPDATE_MEMBER_USERNAME"
+    case cloudOrganizationUpdateMemberRole = "TUIST_CLOUD_ORGANIZATION_UPDATE_MEMBER_ROLE"
+    case cloudOrganizationUpdateMemberPath = "TUIST_CLOUD_ORGANIZATION_UPDATE_MEMBER_PATH"
+
+    // CLOUD AUTH
+    case cloudAuthPath = "TUIST_CLOUD_AUTH_PATH"
```
Comment (pepicrft) [inline Sources/TuistKit/Commands/EnvKey/EnvKey.swift:208]: Some flags and arguments have the same semantics across commands, like in the case of a path. What if, in those cases we reuse the environment variable? For example, we could use `TUIST_PATH` for all the `--path` options.

File: Sources/TuistKit/Commands/EnvKey/ParseableArgument+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,221 @@
+import ArgumentParser
+import Foundation
+import TuistSupport
+
+extension Option {
```
Comment (pepicrft) [inline Sources/TuistKit/Commands/EnvKey/ParseableArgument+Extension.swift:5]: This is awesome! I'd consider opening an issue on the argument parser repository proposing this as a built-in feature of the package.

File: Sources/TuistKit/Commands/GenerateCommand.swift [hunk 3]
```diff
@@ -37,15 +38,16 @@ public struct GenerateCommand: AsyncParsableCommand, HasTrackableParameters {
 
     @Flag(
         name: .shortAndLong,
-        help: "Don't open the project after generating it."
+        help: "Don't open the project after generating it.",
+        envKey: .generateOpen
     )
-    var noOpen: Bool = false
+    var open: Bool = true
```
Comment (pepicrft) [inline Sources/TuistKit/Commands/GenerateCommand.swift:44]: This is a breaking change. I'd consider adopting [the flag inversion](https://swiftpackageindex.com/apple/swift-argument-parser/1.4.0/documentation/argumentparser/flaginversion) option supported by argument parser, and have `TUIST_GENERATE_OPEN=1/0` (truthy == open / falsy == no open)

This comment applies to all the breaking changes in commands.

**Review by pepicrft — COMMENTED** [review]
This is an amazing contribution @ajkolean. Thanks for getting the CLI commands' documentation into our generated static site too.
Besides my comment on breaking changes, we'd also need to get the lining fixed. You can run `mise run lint` locally and `mise run lint-fix` to try to fix it automatically.

**Comment by pepicrft** [comment]
@all-contributors add @ajkolean for code

File: Sources/TuistKit/Commands/EnvKey/EnvKey.swift (hunk 1 above)
Comment (ajkolean) [reply Sources/TuistKit/Commands/EnvKey/EnvKey.swift:208]: It's challenging to strike a balance between granularity and simplicity. The `Test`, `Build`, and `Run` commands each have `--path` options, and there are scenarios where these should be distinct. A shared environment variable could create confusing side effects; for example, setting the build path would unintentionally set it for all commands, which might not be immediately obvious to users.

I opted for extreme explicitness as the default but am open to suggestions for a more consumer-friendly approach.

For consistency, I added an environment variable to every option, flag, and argument. However, some scenarios don't make sense for environment variables, such as `TUIST_CLOUD_ORGANIZATION_INVITE_EMAIL`. It's unlikely anyone would set that. If there's a heuristic to help developers determine what should or shouldn't be an environment variable, that would be useful. I chose to include everything to provide flexibility for consumers and reduce the cognitive load on maintainers by establishing a clear pattern.

**Comment by ajkolean** [comment]
> This is an amazing contribution @ajkolean. Thanks for getting the CLI commands' documentation into our generated static site too. Besides my comment on breaking changes, we'd also need to get the lining fixed. You can run `mise run lint` locally and `mise run lint-fix` to try to fix it automatically.



> This is an amazing contribution @ajkolean. Thanks for getting the CLI commands' documentation into our generated static site too. Besides my comment on breaking changes, we'd also need to get the lining fixed. You can run `mise run lint` locally and `mise run lint-fix` to try to fix it automatically.

I ran `mise run lint-fix`, however seems like few issues that likely require manual intervention, on it. Regarding the CLI commands documentation: it performs a rough parse of the man pages, which is supported by a plugin in the argument parser, similar to its support for auto-completions (`swift package plugin generate-manual`).

That being said, this might be out of scope for this PR. I could move it to a different PR and improve the formatting. Man pages follow a consistent style and have the potential to provide a clear, maintainable source of truth for the CLI documentation, as they are based on the argument parser's command definitions.

File: Sources/TuistKit/Commands/GenerateCommand.swift (hunk 3 above)
Comment (pepicrft) [reply Sources/TuistKit/Commands/GenerateCommand.swift:44]: Well thought of 👏🏼 . I completely missed that in the changes.

File: Sources/TuistKit/Commands/EnvKey/EnvKey.swift (hunk 1 above)
Comment (pepicrft) [reply Sources/TuistKit/Commands/EnvKey/EnvKey.swift:208]: I think your rationale and trade-offs make total sense. Thanks for the clarification.

**Comment by pepicrft** [comment]
Thanks a lot for contributing this one @ajkolean 🙏🏼

**Pull request merged**